src/api.js

```diff
@@ -72,6 +74,10 @@ const generateResponse = (err, resp, rb, options) => {
   }
   if (get(resp, 'isApiResponse') === true) {
     const headers = Object.assign({}, options.defaultHeaders, resp.headers);
+    if (get(resp, 'fields', null) !== null) {
+      assert(get(resp, 'isJsonResponse') === true, 'Can only auto prune JsonResponse.');
+      objectRewrite({ retain: resp.fields })(resp.payload);
```

Comment:
We better don't do anything with the payload past this point... it will be overwritten... is this potentially a problem?